include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,256 @@
+// Copyright (C) 2020 - AnastaZIuk
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_C_SUMMED_AREA_TABLE_IMAGE_FILTER_H_INCLUDED__
+#define __IRR_C_SUMMED_AREA_TABLE_IMAGE_FILTER_H_INCLUDED__
+
+#include "irr/core/core.h"
+
+#include <type_traits>
+#include <functional>
+
+#include "irr/asset/filters/CMatchedSizeInOutImageFilterCommon.h"
+#include "CConvertFormatImageFilter.h"
+
+namespace irr
+{
+namespace asset
+{
+
+template<bool ExclusiveMode>
+class CSummedAreaTableImageFilterBase
+{
+	public:
+		class CSummStateBase
+		{
+			public:
+				enum E_SUM_MODE
+				{
+					ESM_INCLUSIVE,	//!< all the values are summed withing the pixel summing begin from, so (x,y,z) values <= than itself
+					ESM_EXCLUSIVE,	//!< all the values are summed without the pixel summing begin from, so (x,y,z) values < than itself
+					EAS_COUNT
+				};
+
+				uint8_t*	scratchMemory = nullptr;										//!< memory used for temporary filling within computation of sum values
+				uint32_t	scratchMemoryByteSize = 0u;									
+				const E_SUM_MODE mode = ExclusiveMode ? ESM_EXCLUSIVE : ESM_INCLUSIVE;
```

Comment:
not sure why nwe need this and the enum?

Reply:
okay, to remove